src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,134 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> temp = tail;
+        Node<T> newNode = new Node<>(temp, value, null);
+        tail = newNode;
+        if (temp == null) {
+            head = newNode;
+        } else {
+            temp.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        final Node<T> curNode = getNode(index);
+        final Node<T> prevNode = curNode.prev;
+        final Node<T> newNode = new Node<>(prevNode, value, curNode);
+        if (prevNode == null) {
+            head = newNode;
+        } else {
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element: list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        final Node<T> curNode = getNode(index);
+        final T oldValue = curNode.value;
+        curNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
```

Comment:
In this homework for remove methods you should create 'unink' method
![image](https://user-images.githubusercontent.com/117547096/212539959-9091e7a3-40fc-46a3-9f2f-22dec1cb226a.png)